File: Assignments/A1/Biggest.java
import java.io.*;

public class Biggest
{
    public static void main(String[] args)
    {
        BufferedReader fileIn;
        PrintWriter fileOut;
        String inputLine;
        String currWord = "";
        String bigWord = null;
        int bigWeight = 0;
        int weight = 0;
        int index;
        int line = 0;
        
        try
        {
            fileIn = new BufferedReader(new FileReader("in.txt"));
            fileOut = new PrintWriter(new FileWriter("out.txt"));
            
            inputLine = fileIn.readLine();
            while ( inputLine != null )
            {
                index = 0;
                line++;
                bigWord = null;
                bigWeight = 0;
                
                while (index < inputLine.length()) 
                {
                    weight = 0;
                    currWord = "";
                    while (index < inputLine.length() && 
                           (inputLine.charAt(index) != ' ' &&
                            inputLine.charAt(index) != '\t'))
                    {
                        currWord += inputLine.charAt(index);
                        weight += inputLine.charAt(index);
                        index++;
                    }
                    
                    if (weight > bigWeight)
                    {
                        bigWord = currWord;
                        bigWeight = weight;
                    }
                    
                    index++;
                }
                
                if (bigWord != null)
                {
                    fileOut.print( "line " + line + "'s word: " + bigWord );
                    fileOut.println();
                }
                
                inputLine = fileIn.readLine();
            }
            
            
            fileIn.close();
            fileOut.close();
        }
        
        catch (IOException ioe)
        {
            System.out.println(ioe.getMessage());
            ioe.printStackTrace();
        }
    }
}

File: Assignments/A1/biggest.c
#include <stdio.h>
#include <string.h>

#define LINE_MAX 100

int main(int argc, char *argv[])
{
	char inputLine[LINE_MAX];

	char currWord[LINE_MAX];
	char bigWord[LINE_MAX];

	int bigWeight = 0;
	int weight = 0;
	int index;
	int line = 0;

	fgets(inputLine, LINE_MAX, stdin);

	while(scanf("%c",inputLine) != EOF)
	{

		index = 0;

		line++;
		memset(bigWord,0,strlen(bigWord));
		bigWeight = 0;

		while (index < strlen(inputLine))
		{
			weight = 0;
			memset(currWord, 0, strlen(currWord));
			int currWordIndex = 0;
			while (index < strlen(inputLine) && (
					(inputLine[index] != ' ') &&
					(inputLine[index] != '\t')))
			{
				/*inputPtr has address of inputLine[],
				 *so *inputPtr will take that address and fetch data from it.
				*/
				currWord[currWordIndex] = inputLine[index];
				weight += inputLine[index]; //ASCII value

				index++;
			}
			if(weight > bigWeight)
			{
				strcpy(bigWord,currWord);
				bigWeight = weight;
			}
			index++;
		}
		if (strcmp(bigWord, " ") != 0)
		{
			printf("\nline %d's word: %s", line, bigWord);
		}
		fgets(inputLine, LINE_MAX, stdin);
//		printf("Next INPUT: %s",inputLine);
	}
}
